Basics/Git.md

```diff
@@ -4,50 +4,130 @@
+In general, commits should be small and frequent - “Commit early, commit often”. You should follow the single responsibility principle in git. Whenever possible include task number in commit message, e.g.:”#123: Fix screen title”, this helps a lot in the future when another person needs to know why you had to implement it that way.
 
-At any given point in time, only one developer should be actively working on one of these branches.
+The important thing is that changes should be described in imperative mood, e.g. "Make xyzzy do frotz" instead of "[This patch] makes xyzzy do frotz" or "[I] changed xyzzy to do frotz", as if you are giving orders to the codebase to change its behavior.
 
-Once more, you should submit a pull request to *develop* only when everything is tested, working, and not breaking the project. Once it has been approved, close your branch down and start working on a new one.
+When writing commit messages we should stick to these 7 rules:
+1. Separate subject from body with a blank line
+2. Limit the subject line to 50 characters (easier to read on client)
+3. Capitalize the subject line
+4. Do not end the subject line with a period
+5. Use the imperative mood in the subject line
+6. Wrap the body at 72 characters
+7. Use the body to explain what and why vs. how
+8. You can find more details about those in [this blog post][3].
 
-## 3. GitFlow for small projects
+There is also a [git hook][4] available to validate your commit message.
 
-In order to keep things rolling faster, small projects also use GitFlow, but to a very limited extent.
+## 6. Pull requests
 
-For one-man projects, you will have only the **master** and **develop** branches, and will commit your code directly to develop, while the master stays the same.
+The preferred way of merging a branch back into main is by creating a pull request and assigning it to a colleague for review. At least one person should review any code that is going to be merged (reviewer on PR).
 
-If there is more than one developer, you'll need a standard GitFlow model.
+Each PR should have a description. Link to the Productive/Jira task is minimum for description. It would be great to include a link to design (if applicable) and describe the changes that PR introduces. You can find a good read on what the description should contain [here][5].
 
-## 4. Tools (optional)
+PR should have an assignee, which is usually a person who created a PR or the person who will be responsible for merging it. 
 
-Although you don’t need to use any tools with GitFlow workflow, and may prefer sticking to the workflow manually, you still have an option to add commands to your console/terminal for commonly used workflows.
+Master and/or develop branches should always be marked as protected on Github in order to prevent direct push to those branches. Next options should be checked on branch protection rules page on Github:
```

Comment:
```suggestion
Master and/or release branches should always be marked as protected on Github in order to prevent direct push to those branches. Next options should be checked on the branch protection rules page on Github:
```